### ISSUE TITLE
[BUG] Some photos year more than 4 digit doesn't show in timeline

### ISSUE THREAD

**Issue by harshjk** [immich-app/immich#3527]
### The bug

[BUG] I got some forwarded photos and videos in Whatsapp and I see that the year of these media are more than 4 digits. So It'll not appear in the timeline. I attached photos here.
<img width="540" alt="image" src="https://github.com/immich-app/immich/assets/3693920/9f013da0-ec2a-4267-b16d-2a902f9a5b83">
<img width="1722" alt="image" src="https://github.com/immich-app/immich/assets/3693920/86a12d66-f07d-4063-9955-55fcfa08645e">

### The OS that Immich Server is running on

Debian

### Version of Immich Server

v1.70

### Version of Immich Mobile App

v1.71

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Same as v1.70 docker-compose.yml
```

### Your .env content

```Shell
Same as v1.70 example.env and just update the upload path
```

### Reproduction steps

```bash
1. try to upload photos with year more than 4 digits in that like 123456
```

**Comment by uhthomas** [comment]
Yeah, this is a bit weird but I don't think we have any reason to support photos from the year 10,000 or beyond.

**Comment by uhthomas** [comment]
I guess thinking further we should try to understand why the year on that photo was so wrong, and maybe handle that better? Was the image uploaded from the app?

**Comment by uhthomas** [comment]
Maybe https://github.com/immich-app/immich/issues/3547?

**Comment by harshjk** [comment]
> I guess thinking further we should try to understand why the year on that photo was so wrong, and maybe handle that better? Was the image uploaded from the app?

Most are forwarded photos and videos. No, It doesn't upload from the app. I used bulk upload docker CLI.

**Comment by gmag11** [comment]
Whatsapp pictures management is shit. In this case date is indicated on filename. For instance `IMG-20200210-WA0000.jpg` corresponds with 2020/feb/10. All metadata into file, if any, should be ignored.

**Comment by samip5** [comment]
Whatsapp seems to be scrampling the EXIF on the file, so yes, metadata should be ignored if there is any.
I think parsing the filename would be an option.

**Comment by uhthomas** [comment]
Would anyone be kind enough to show the full output of exiftool on a WhatsApp image?

**Comment by Fritzip** [comment]
`exiftool IMG-20230618-WA0001.jpg`

```
ExifTool Version Number         : 12.60
File Name                       : IMG-20230618-WA0001.jpg
Directory                       : .
File Size                       : 138 kB
File Modification Date/Time     : 2023:06:18 14:52:50+02:00
File Access Date/Time           : 2023:06:18 14:52:50+02:00
File Inode Change Date/Time     : 2023:09:07 12:01:26+02:00
File Permissions                : -rw-r--r--
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
JFIF Version                    : 1.01
Resolution Unit                 : None
X Resolution                    : 1
Y Resolution                    : 1
Image Width                     : 1200
Image Height                    : 1600
Encoding Process                : Progressive DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Image Size                      : 1200x1600
Megapixels                      : 1.9
```

**Comment by erikvanoosten** [comment]
The same happens with photos that are shared via Signal: all exif data is gone.
My image import pipeline corrects for this by parsing the filename 🤷

**Comment by gmag11** [comment]
Just for information.

Librephotos did a hacky solution for pictures with no metadata https://docs.librephotos.com/docs/user-guide/settings/date-rules

It has a list of defined rules to extract dates from different places (standar exif tag, alternate tags, filename regex, etc). User may add some of that rules from a list and may assign different priority to each rule.

![image](https://github.com/immich-app/immich/assets/6190183/6ae85b65-48eb-4c7c-aaa6-235bfb07c8d6)

**Comment by icecore2** [comment]
The issue still there, the parsing not works properly or at all.
Librephotos solution sounds nice, let the user build own parsing methods with built-in ones.

**Comment by ach-raf** [comment]
if I edit an image manualy or using a script, to modify and fix the date for whatsapp images, how can I make immich take the new changes on considiration, I even moved the image to the correct folder structure in my immich library, but it still shows as the old date.

**Comment by moseshewetson** [comment]
I have just discovered this issue after uploading around 50,000 images to a library. Maybe 20% of them are Whatsapp images that have the dates all wrong! Not I need to somehow bulk delete them from my immich library if anyone has any ideas.

**Comment by alextran1502** [comment]
Moving to feature request
